oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileSplitterTest.java

```diff
@@ -533,4 +548,15 @@ public void uncompress(File src, File dest) throws IOException {
             }
         }
     }
+    
+    private static Set<String> readAllFilesAsSet(List<File> files) {
+        Set<String> set = files.stream().flatMap(file -> {
+            try ( InputStream is = Files.newInputStream(file.toPath())) {
```

Comment:
```suggestion
            try (InputStream is = Files.newInputStream(file.toPath())) {
```

Reply:
> I still see leftover files in TMPDIR, but we can handle that as a less pressing, separate issue.

This should be fixed with my latest commits to
- Creating TemporaryFolder into the target directory
- Setting the system property java.tmpdir under the temporary folder

Reply:
Hm. My main concern is not leftover files from tests, but unclosed files in general. These may not manifest themselves as leftover files on non-Windows systems, but would still be a problem in production.

Reply:
Yes good point. I am checking that too.